### PR TITLE
yolov8 notebook

### PR DESCRIPTION
**Pull request by eaidova**

### CODE REVIEW

**Comment by raymondlo84** [comment]
Wow YoloV8.

I wonder if we can make this a webcam demo?

**Comment by eaidova** [comment]
> Hi Since YOLO's input and out layout are almost same, I just wonder can we combine all YOLO-based instance into one notebook?

I think it is not good idea at least for now, when they have a pick of popularity. People search such tutorials and want to have only related stuff. Downloading and converting approaches are different, it will lead to mess (however it is usual state for yolo) and do not provide more clarity...
Beside that this tutorial not fibnished in quantization part, I got some issues with model accuracy, so this part I believe requires more efforts then for other models. I have nothing against to combain it with yolov5, because now ultralytics provides unified api for both yolov8 and yolov5.

**Comment by MaximProshin** [comment]
> > Hi Since YOLO's input and out layout are almost same, I just wonder can we combine all YOLO-based instance into one notebook?
> 
> I think it is not good idea at least for now, when they have a lot of popularity. People search such tutorials and want to have only related stuff. Downloading and converting approaches are different, it will lead to mess (however it is usual state for yolo) and do not provide more clarity... Beside that this tutorial not finished in quantization part, I got some issues with model accuracy, so this part I believe requires more efforts then for other models. I have nothing against to combain it with yolov5, because now ultralytics provides unified api for both yolov8 and yolov5.

One option could be to have a common notebook which introduces to YOLO and compares it's different versions which are generated in separate sub-notebooks for each version. That notebook could be useful if the user wants to compare accuracy and performance of different version on the same HW and even same video sample.

**Comment by paularamo** [comment]
> > > Hi Since YOLO's input and out layout are almost same, I just wonder can we combine all YOLO-based instance into one notebook?
> > 
> > 
> > I think it is not good idea at least for now, when they have a lot of popularity. People search such tutorials and want to have only related stuff. Downloading and converting approaches are different, it will lead to mess (however it is usual state for yolo) and do not provide more clarity... Beside that this tutorial not finished in quantization part, I got some issues with model accuracy, so this part I believe requires more efforts then for other models. I have nothing against to combain it with yolov5, because now ultralytics provides unified api for both yolov8 and yolov5.
> 
> One option could be to have a common notebook which introduces to YOLO and compares it's different versions which are generated in separate sub-notebooks for each version. That notebook could be useful if the user wants to compare accuracy and performance of different versions on the same HW and even same video sample.

Maybe we could create a series of notebooks over the same number(folder) and include several approaches. I found it interesting in this notebook, that is covering the complete pipeline conversion, optimization, validation and also using the benchmark app. I will add more comments in my review.

**Comment by raymondlo84** [comment]
> Hi Since YOLO's input and out layout are almost same, I just wonder can we combine all YOLO-based instance into one notebook?

We can build a webcam demo that can toggle between all these models maybe? How do we host the pre-trained and optimized model without model zoo? 

I think it's great if we can run these demo side by side, and then reference back to the 'how-to-get-there' notebooks here.

**Comment by paularamo** [comment]
Adding this to my previous review:
An error in cell 10.
image
In my first review I used a 10th Gen Machine - Windows 10 - Python 3.8.0

This error is happening in a 12th Gen Machine - Windows 11 - Python 3.9.13

**Comment by OpenVINO-dev-contest** [comment]
> > Hi Since YOLO's input and out layout are almost same, I just wonder can we combine all YOLO-based instance into one notebook?
> 
> We can build a webcam demo that can toggle between all these models maybe? How do we host the pre-trained and optimized model without model zoo?
> 
> I think it's great if we can run these demo side by side, and then reference back to the 'how-to-get-there' notebooks here.

Yes, But we have already got 401 for object detection webcam ?  shall we put them together in it ? or let's re-design our notebooks category. For example, put all objection model example into one separate fold and label it for different use case (entry level, webcam, model optimization or solution level)?

**Comment by eaidova** [comment]
> Adding this to my previous review: An error in cell 10. image In my first review I used a 10th Gen Machine - Windows 10 - Python 3.8.0
> 
> This error is happening in a 12th Gen Machine - Windows 11 - Python 3.9.13

OMG! They change interface again.  :(
Probably issue connected that you use 8.0.5 version (in 8.0.3 format was different from 8.0.4 too), thanks I take a look

**Comment by eaidova** [comment]
<p>Regarding loggin, it looks like some difference logging setup on windows and linux machine (it is expected at least in emoji usage part, because their printing is problematic. Regarding return none, please see https://github.com/ultralytics/ultralytics/issues/313</p><p>As I already answered on Paula comment, they change api. I pin yolov8 version for now for avoid unpredictable result</p><p> </p>

---

 <a href='https://app.reviewnb.com/openvinotoolkit/openvino_notebooks/pull/776/discussion/'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

**Comment by eaidova** [comment]
<p>misclick, removed</p><p> </p>

---

 <a href='https://app.reviewnb.com/openvinotoolkit/openvino_notebooks/pull/776/discussion/'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

**Comment by eaidova** [comment]
<p>added</p>

---

 <a href='https://app.reviewnb.com/openvinotoolkit/openvino_notebooks/pull/776/discussion/'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

**Comment by eaidova** [comment]
<p>added</p>

---

 <a href='https://app.reviewnb.com/openvinotoolkit/openvino_notebooks/pull/776/discussion/'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

**Comment by eaidova** [comment]
<p>increased output resolution</p>

---

 <a href='https://app.reviewnb.com/openvinotoolkit/openvino_notebooks/pull/776/discussion/'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

**Comment by eaidova** [comment]
<p>added explanation how to read this plot</p>

---

 <a href='https://app.reviewnb.com/openvinotoolkit/openvino_notebooks/pull/776/discussion/'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

**Comment by eaidova** [comment]
<p>now they should be consecutive</p>

---

 <a href='https://app.reviewnb.com/openvinotoolkit/openvino_notebooks/pull/776/discussion/'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

**Comment by eaidova** [comment]
<p>provided</p>

---

 <a href='https://app.reviewnb.com/openvinotoolkit/openvino_notebooks/pull/776/discussion/'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

**Comment by eaidova** [comment]
<p>added</p>

---

 <a href='https://app.reviewnb.com/openvinotoolkit/openvino_notebooks/pull/776/discussion/'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

**Comment by eaidova** [comment]
<p>aligned</p><p> </p>

---

 <a href='https://app.reviewnb.com/openvinotoolkit/openvino_notebooks/pull/776/discussion/'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

**Comment by AlexKoff88** [comment]
> > > Hi Since YOLO's input and out layout are almost same, I just wonder can we combine all YOLO-based instance into one notebook?
> > 
> > 
> > I think it is not good idea at least for now, when they have a lot of popularity. People search such tutorials and want to have only related stuff. Downloading and converting approaches are different, it will lead to mess (however it is usual state for yolo) and do not provide more clarity... Beside that this tutorial not finished in quantization part, I got some issues with model accuracy, so this part I believe requires more efforts then for other models. I have nothing against to combain it with yolov5, because now ultralytics provides unified api for both yolov8 and yolov5.
> 
> One option could be to have a common notebook which introduces to YOLO and compares it's different versions which are generated in separate sub-notebooks for each version. That notebook could be useful if the user wants to compare accuracy and performance of different versions on the same HW and even same video sample.

I am thinking that this is probably not a good idea. With this, we show the advantages of some models to the user while creating a negative perception of other models. You should hear about the strong competition between YOLOv7 and YOLOv5 Ultralytics which led to YOLOv8 finally. I would suggest not making comparisons of algorithms, models, etc. but letting the user decide on his end based on the public data or whatever.

**Comment by eaidova** [comment]
<p>this resize type in not supported, because it is preserves aspect ratio plus requires padding.</p><p>also not sure about dims expanding (at least on python side HWC layout can not be fit to NCHW)</p>

---

 <a href='https://app.reviewnb.com/openvinotoolkit/openvino_notebooks/pull/776/discussion/'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

**Comment by eaidova** [comment]
<p>I can rewrite this op on numpy for now instead of using it on yolo side</p><p> </p>

---

 <a href='https://app.reviewnb.com/openvinotoolkit/openvino_notebooks/pull/776/discussion/'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

**Comment by paularamo** [comment]
FP32 - 330FPS with ARC A770

image

INT8 - 356 FPS with ARC 770

image

Nice!

**Comment by paularamo** [comment]
Webcam - OV_model FP32 - 12Gen Machine
image

Webcam - INT8 model - 12Gen Machine
image

My best pajama version 😂

**Comment by paularamo** [comment]
@eaidova, could you please add the device_name option? so we will have the flexibility to change the device if needed. Thanks

**Comment by eaidova** [comment]
<p>fixed, thanks</p><p> </p>

---

 <a href='https://app.reviewnb.com/openvinotoolkit/openvino_notebooks/pull/776/discussion/'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

**Comment by eaidova** [comment]
<p>agree, changed structure</p>

---

 <a href='https://app.reviewnb.com/openvinotoolkit/openvino_notebooks/pull/776/discussion/'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

**Comment by eaidova** [comment]
<p>I deeply analyze postprocessing, it will be a lot of code if I add it directly to notebook... I think we can have it in future or at least it can be possible to converted using pytorch fe (torchvision.nms layer is already in plans, yolov8 nms based on it)</p>

---

 <a href='https://app.reviewnb.com/openvinotoolkit/openvino_notebooks/pull/776/discussion/'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

**Comment by eaidova** [comment]
<p>I added next steps section with info how to use async api and integrate preprocessing (but letterbox stayed in user code), I think we can consider it as part of preprocessing api</p>

---

 <a href='https://app.reviewnb.com/openvinotoolkit/openvino_notebooks/pull/776/discussion/'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

**Comment by paularamo** [comment]
image
@eaidova Do you know how it is possible to fix this? It is just working on CPU, iGPU and dGPU are not working. :(

**Comment by eaidova** [comment]
> image @eaidova Do you know how it is possible to fix this? It is just working on CPU, iGPU and dGPU are not working. :(
> 
> `run_object_detection(source=0, flip=True, use_popup=False, model=ov_model, device="CPU")` is the function to work with the webcam and see the inference in real-time.

@paularamo probably ov_model should be reshaped to have static shape before loading- for working on GPU as it does not support dynamic shapes.
we need to have it dynamic for obtaining accuracy results (inside yolov8 code, it uses preprocessing which lead to dynamic input shapes

**Comment by paularamo** [comment]
image

detect_without_preprocess and quantize_model_with_preprocess is much faster but I cannot see correct BB over the video.
image

**Comment by paularamo** [comment]
Thanks @eaidova It is working with GPUs now.

image

**Comment by eaidova** [comment]
<p>done</p><p> </p>

---

 <a href='https://app.reviewnb.com/openvinotoolkit/openvino_notebooks/pull/776/discussion/'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

**Comment by raymondlo84** [comment]
image

Since we are sharing the dataset, shouldn't this be part of the data directory @ the root folder?

**Comment by raymondlo84** [comment]
image
impressive... A770 and YoloV8 = over 1000fps.

**Review by ryanloney — APPROVED** [review]

**Comment by yangsu2022** [comment]
Hi @eaidova ,nice work! 
Any plans for the yolov8's segmentation?
